docs/GuidGuidRecommender.md

```diff
@@ -0,0 +1,144 @@
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+it is supplied as `tie_breaker_dict`.
+It is also used as the `ranking_dict` passed to the [`MinInstallPrune`](../taar_lite/recommenders/treatments.py#L36) treatment.
+
+
+# Generating the scored candidate lists
+
+The TAAR-lite recommender is built on the following underlying principle:
+_add-ons which tend to be installed together are similar or complimentary,
+and the number of coinstallations indicates the strength of this relationship._
+Thus, starting from a given add-on, good recommendations for other add-ons
+are those which are most often coinstalled.
+
+Given an add-on A, a natural choice for a candidate list and relevance scores
+is then just the list of add-ons that are coinstalled with A,
+together with their coinstallation counts.
+However, recommendation quality can be improved by making certain adjustments
+to these preliminary scored lists, such as:
+
+- tempering the influence of highly popular add-ons
+    so that they don't drown out more relevant, less common ones
+- pruning out obscure add-ons
+- better balancing relevance against diversity
+- improving the experience for users clicking through a chain of recommendations
+    on AMO, such as pruning out cyclical recommendations.
+
+Final candidate lists and relevance scores are generated
+by applying a sequence of such transformations,
+referred to as __treatments__,
+to the raw coinstallation data.
+As recommendations are based solely on the treated data,
+we are assuming that it encapsulates all the information
+necessary in generating recommendations.
+Treatments are discussed in more detail in terms of their graph representations
+[here](./GraphRepresentation.md#treatments).
+
+Treatments are implemented as extensions of the [`BaseTreatment`](../taar_lite/recommenders/treatments.py#L11) class,
+which apply a transformation via the `treat()` method.
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+a list of `BaseTreatment` instances are supplied as `treatments`.
+When the treated dataset is built, the treatments are applied in sequence
+to the raw coinstallation data,
+and the result is stored as the recommender instance's `treated_graph`.
+Candidate lists and relevance scores are then drawn
+from the treated dataset.
+
+
+# Selecting recommendations
+
+Recommendations for a given add-on A are generated
+by ordering A's candidate list by decreasing relevance score
+and selecting the first N add-ons from the sorted list.
+Ties in relevance score are broken using an additional list
+of add-on tie-breaking scores,
+by further ordering tied candidates by decreasing tie-breaking score.
+
+Recommendations are requested from a `GuidGuidCoinstallRecommender` instance
+using the [`recommend()`](../taar_lite/recommenders/guidguid.py#L123) method.
+The list of tie-breaking scores was supplied on initialization as `tie_breaker_dict`.
```

Comment:
I would suggest cutting from here to the end - see previous comments about `tie_breaker_dict` and deployment vs class concerns.